runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/EngineManager.java

```diff
@@ -421,31 +421,33 @@ private void unregister(
     private final class NameResolver
     {
         private final int namespaceId;
-        private final Matcher matchName;
+        private final ThreadLocal<Matcher> matchName;
 
         private NameResolver(
             int namespaceId)
         {
             this.namespaceId = namespaceId;
-            this.matchName = NamespaceAdapter.PATTERN_NAME.matcher("");
+            this.matchName = new ThreadLocal<>();
         }
 
         private long resolve(
             String name)
         {
             long id = 0L;
-
-            if (name != null && matchName.reset(name).matches())
+            if (name != null)
             {
-                String ns = matchName.group("namespace");
-                String n = matchName.group("name");
+                matchName.set(NamespaceAdapter.PATTERN_NAME.matcher(name));
```

Comment:
This will create a new instance and set the thread local value on every call to resolve method, defeating the purpose of the thread local, to safely _reuse_ the matcher.

Instead use `ThreadLocal.withInitial(...)` as suggested in the constructor above.